app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt

```diff
@@ -169,26 +182,75 @@ class MessagingService : FirebaseMessagingService() {
 
         handleImage(notificationBuilder, data)
 
-        handleActions(notificationBuilder, data, tag, messageId)
+        handleActions(notificationBuilder, tag, messageId, data)
+
+        handleDeleteIntent(notificationBuilder, messageId, group, groupId)
+
+        handleContentIntent(notificationBuilder, messageId, group, groupId, data)
 
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
             handleLegacyPriority(notificationBuilder, data)
             handleLegacyLedColor(notificationBuilder, data)
             handleLegacyVibrationPattern(notificationBuilder, data)
         }
 
-        notificationManagerCompat.apply {
+        notificationManager.apply {
             notify(tag, messageId, notificationBuilder.build())
             if (group != null) {
                 notify(group, groupId, getGroupNotificationBuilder(channelId, group, data).build())
+            } else {
+                if (!previousGroup.isNullOrBlank()) {
+                    notificationManager.cancelGroupIfNeeded(previousGroup, previousGroupId)
+                }
```

Comment:
If we are adding a notification why would we ever need to cancel the group?

Reply:
If the notification was in a group and it is updated (by the tag property) and it was taken out of the  group by removing the group property, the left empty group needs to be removed (previousGroup).

So first it updates the notification with the notify method, then we check if we had a group in the last notification. (previousGroup). If yes we cancel the group, if its empty.

Reply:
Ok, makes sense.  Think it's a super edge case in practice but good catch.